stan/math/prim/mat/fun/autocorrelation.hpp

```diff
@@ -114,24 +114,23 @@ void autocorrelation(const Eigen::MatrixBase<DerivedA>& y,
   size_t M = fft_next_good_size(N);
   size_t Mt2 = 2 * M;
 
-  Eigen::Matrix<std::complex<T>, Eigen::Dynamic, 1> freqvec;
-
   // centered_signal = y-mean(y) followed by N zeros
   Eigen::Matrix<T, Eigen::Dynamic, 1> centered_signal(Mt2);
   centered_signal.setZero();
   centered_signal.head(N) = y.array() - y.mean();
 
+  Eigen::Matrix<std::complex<T>, Eigen::Dynamic, 1> freqvec(Mt2);
   fft.fwd(freqvec, centered_signal);
   // cwiseAbs2 == norm
   freqvec = freqvec.cwiseAbs2();
 
-  fft.inv(ac, freqvec);
-  ac.derived().conservativeResize(N);
+  Eigen::Matrix<std::complex<T>, Eigen::Dynamic, 1> ac_tmp(Mt2);
+  fft.inv(ac_tmp, freqvec);
 
   for (size_t i = 0; i < N; ++i)
-    ac(i) /= (N - i);
-  T var = ac(0);
-  ac = ac.array() / var;
+    ac_tmp(i) /= (N - i);
+
+  ac = ac_tmp.head(N).real().array() / ac_tmp(0).real();
```

Comment:
@bob-carpenter This line silently fails the unit tests whenever this function is called with `Eigen::Map` wrapping a `std::vector`, but I can't figure out why as no error messages show up.  If you have any time to provide some thoughts, I'd much appreciate it.